logstash-core/src/main/java/org/logstash/health/PipelineIndicator.java

```diff
@@ -0,0 +1,195 @@
+                gen.writeStringField("state", value.getState().toString());
+                gen.writeEndObject();
+            }
+        }
+    }
+
+    @JsonSerialize(using = Details.JsonSerializer.class)
+    public static class Details implements Observation {
+        private final Status status;
+
+        public Details(final Status status) {
+            this.status = Objects.requireNonNull(status, "status cannot be null");
+        }
+        public Status getStatus() {
+            return this.status;
+        }
+
+        public static class JsonSerializer extends com.fasterxml.jackson.databind.JsonSerializer<Details> {
+            @Override
+            public void serialize(final Details details,
+                                  final JsonGenerator jsonGenerator,
+                                  final SerializerProvider serializerProvider) throws IOException {
+                jsonGenerator.writeStartObject();
+                jsonGenerator.writeObjectField("status", details.getStatus());
+                jsonGenerator.writeEndObject();
+            }
+        }
+    }
+
+    /**
+     * This interface is implemented by the ruby-Agent
+     */
+    @FunctionalInterface
+    public interface PipelineDetailsProvider {
+        Details pipelineDetails(final String pipelineId);
+    }
+
+    public static class DetailsSupplier {
+        private final String pipelineId;
+        private final PipelineDetailsProvider pipelineDetailsProvider;
+        DetailsSupplier(final String pipelineId,
+                        final PipelineDetailsProvider pipelineDetailsProvider) {
+            this.pipelineId = pipelineId;
+            this.pipelineDetailsProvider = pipelineDetailsProvider;
+        }
+
+        public Details get() {
+            return this.pipelineDetailsProvider.pipelineDetails(pipelineId);
+        }
+    }
+
+    static class StatusProbe implements Probe<Details> {
+        static final Impact.Builder NOT_PROCESSING = Impact.builder()
+                .withId(impactId("not_processing"))
+                .withDescription("the pipeline is not currently processing")
+                .withAdditionalImpactArea(ImpactArea.PIPELINE_EXECUTION);
+
+        static final HelpUrl HELP_URL = new HelpUrl("health-report-pipeline-status");
+
+        @Override
+        public Analysis analyze(final Details details) {
+            switch (details.getStatus().getState()) {
+                case LOADING:
+                    return Analysis.builder()
+                            .withStatus(YELLOW)
+                            .withDiagnosis(db -> db
+                                    .withId(diagnosisId("loading"))
+                                    .withCause("pipeline is loading")
+                                    .withAction("if pipeline does not come up quickly, you may need to check the logs to see if it is stalled")
+                                    .withHelpUrl(HELP_URL.withAnchor("loading").toString()))
```

Comment:
With the [doc PR](https://github.com/elastic/logstash/pull/16513), the anchor changed to `health-report-pipeline-status-diagnosis-{status}` style.
```suggestion
                                    .withHelpUrl(HELP_URL.withAnchor("health-report-pipeline-status-diagnosis-loading").toString()))
```

Reply:
I'm trying today in that PR to within the docs subsystem make a second in-page-only anchor, so that we can link directly within the docs to the globally-distinct id but can also link from the outside with shortened id's.

Reply:
I've added in secondary anchors to the docs PR:

 - https://logstash_bk_16513.docs-preview.app.elstc.co/guide/en/logstash/8.16/health-report-pipeline-status.html#finished
 - https://logstash_bk_16513.docs-preview.app.elstc.co/guide/en/logstash/8.16/health-report-pipeline-status.html#loading
 - https://logstash_bk_16513.docs-preview.app.elstc.co/guide/en/logstash/8.16/health-report-pipeline-status.html#terminated
 - https://logstash_bk_16513.docs-preview.app.elstc.co/guide/en/logstash/8.16/health-report-pipeline-status.html#unknown